### ISSUE TITLE
'Recursive' missing from deleteRuleFromRuleList

### ISSUE THREAD

**Issue by nemethf** [l7mp/l7mp#81]
Can you, please, implement something like the following:

```patch
--- a/openapi/l7mp-openapi.yaml
+++ b/openapi/l7mp-openapi.yaml
@@ -1623,6 +1623,13 @@ paths:
               - type: integer
               - type: string
           description: Position or name of the Rule to delete.
+        - name: recursive
+          in: query
+          schema:
+            type: boolean
+          description: >
+            Recursively delete all sub-objects (routes, clusters, etc.) that
+            were implicitly defined (auto-generated) in the Rule.
       responses:
         '200':
           description: Status.
```

In my case, the a (named) route was defined as a part of a rule.

**Comment by rg0now** [comment]
My understanding is that you'd like to have a "recursive" version of `deleteRuleFromRuleList`, is that right? Note that there is a difference between removing a rule from a rulelist and to actually delete it as the latter would make the rule unavailable to the rest of the rulelists as well that refer to it. This difference is reflected in the API. The OpenAPI doc for `deleteRuleFromRuleList` mentions this:

> Note that the Rule itself will not be deleted, just removed from this rulelist; to actually delete the Rule, use the delete operation of the Rule API (deleteRule).

So I'd recommend issuing a separate `deleteRule` with `recursive=true` to the rule name, this would do what you need.

**Comment by nemethf** [comment]
Still, can you add something that avoids multiple calls?

**Comment by rg0now** [comment]
Implemented in PR #83
